### ISSUE TITLE
Is classpath correctly set for a custom connectionType?

### ISSUE THREAD

**Issue by puzzle-it-nu** [jaynath-d/node-jdbc-driver#28]
First, thank you for a more modern version to use jdbc in node :+1: 

I have been evaluating you package with a custom Advantage database jdbc driver, and ran into a problem that as soon as I try to get data from the database than I'll get an error that it can't find the class.

After tracing through the library from my code I noticed that the classpath of the Java instance points to the src folder or the drivers folder of the **installed** package in my node_modules folder. If I copy my jar file to that location then it works but isn't the right thing todo I think.

After another look I see that on line 32 in index.ts the classpath is set:
```js
   jinst.setupClasspath([path.join(__dirname, this.driverPath)]);
```
where `__dirname` is node_modules/node-jdbc-driver/dist and `this.driverPath is ../drivers/adsjdbc-10.0.0.49.jar, that is the  `jars: ` entry from the jdbcDrivers() option object.

I see two ways to solve this
- add enough `../` to my `jars:` entry to escape out of the node_modules folder
- or check for custom type and document that you then need an absolute path to the jdbc.jar file
-  something else I haven't thought about :-)

**Comment by puzzle-it-nu** [comment]
Well, had another look at it and found a more sustainable solution I think.

In the referenced file and line number, change the `__dirname` to `process.cwd()` and then the path in the config will be a reference to the base of you own repo instead of a reference to the base of the installed package in the node_modules folder.

Will make a PR for you.

**Comment by jaynath-d** [comment]
Hi, thanks for raising this issue.

I tested this locally and found the following behavior:
```js
jinst.setupClasspath([path.join(__dirname, this.driverPath)]);
```
- `__dirname` output: `/Users/jaynath-d/Desktop/demo/node-demo/node_modules/node-jdbc-driver/drivers/postgresql-42.6.0.jar`
- `process.cwd()` output: `/Users/jaynath-d/Desktop/demo/drivers/postgresql-42.6.0.jar`

Using `process.cwd()` works for **custom** drivers, but it won't work for others like:

- Hive (`hive`)
- PostgreSQL (`postgresql`)
- SQLite (`sqlite`)
- Tibero (`tibero`)

This is because we only receive the JAR file for **custom** drivers, whereas for the others, the JAR files are already provided and located inside the `node_modules` path.

**Solution for Custom JARs**
To properly pass a **custom** JAR, you can structure your setup like this:
- **Create a `drivers` folder inside your repository** (e.g., inside `node-demo`).
- Use the following configuration to pass the local JAR path:
```js
const jdbc = new JdbcDriver(ConnectionType.custom, {
    jars: '../../../drivers/sqlite-jdbc-3.45.3.0.jar', // Local path of your JAR file
    driverClass: 'org.sqlite.JDBC', // Driver class from the JAR
    jdbcUrl: 'jdbc:sqlite://Users/jaynath-d/Desktop' // JDBC URL for connection
});
console.log("path", path.join(__dirname, _this.driverPath))
```
- **Final JAR path output**: `/Users/jaynath-d/Desktop/demo/node-demo/drivers/sqlite-jdbc-3.45.3.0.jar`

This ensures the JAR is correctly referenced from your project folder rather than `node_modules`.

Let me know if you have any questions!

**Comment by puzzle-it-nu** [comment]
You're correct and I added a check for `custom` or not. Might be my own problem but I'm using `pnpm` to get this compiled.
`npm` version 22 seems to have problems compiling `java` , v20 seems to work. Updating `nan` might also help but using `pnpm` worked for me except for the path problem.

`pnpm test` and `pnpm test:c` are all OK
